File: week-2/distribute-candies/distribute-candies.py
class Solution:
    def distributeCandies(self, candies: List[int]) -> int:
        hashtable = {}
        for i in candies:
            hashtable[i] = hashtable[i] + 1 if i in hashtable else 1
        
        limit = len(candies) //2
        return min(len(hashtable.items()), limit)

File: week-2/distribute-candies/distribute-candies.js
/**
 * @param {number[]} candies
 * @return {number}
 */
var distributeCandies = function(candies) {
    const hashtable = {};
    for (let i of candies) {
        i in hashtable ? hashtable[i]++ : hashtable[i] = 1;
    }
    
    return Math.min(Object.keys(hashtable).length, candies.length/2);
};
